Framework/API/src/ExperimentInfo.cpp

```diff
@@ -110,6 +111,7 @@ void ExperimentInfo::copyExperimentInfoFrom(const ExperimentInfo *other) {
   //   not here since we just copy the experiment data).
   // - For cached groupings (for MDWorkspaces), grouping was not copied in the
   //   old implementation either.
+  *m_componentInfo = *other->m_componentInfo;
```

Comment:
I think actually this could be removed, as well as the line `*m_detectorInfo = *other->m_detectorInfo;` above. `setInstrument()` already handles this, I think?